src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetEnvironmentVariable.cs

```diff
@@ -8,15 +8,7 @@ internal static partial class Interop
 {
     internal static partial class Kernel32
     {
-        internal static unsafe int GetEnvironmentVariable(string lpName, Span<char> buffer)
-        {
-            fixed (char* bufferPtr = &MemoryMarshal.GetReference(buffer))
-            {
-                return GetEnvironmentVariable(lpName, bufferPtr, buffer.Length);
-            }
-        }
-
         [DllImport(Libraries.Kernel32, EntryPoint = "GetEnvironmentVariableW", SetLastError = true, CharSet = CharSet.Unicode, ExactSpelling = true)]
-        private static extern unsafe int GetEnvironmentVariable(string lpName, char* lpBuffer, int nSize);
+        internal static extern unsafe uint GetEnvironmentVariable(string lpName, ref char lpBuffer, uint nSize);
```

Comment:
I am not fan of using `ref char` for buffers. This pattern is used for all other similar methods, so I have used it here too. Hopefully, we will get something better with source-generated interop eventually.

Reply:
Would it be less churn to keep the `GetEnvironmentVariable(string lpName, Span<char> buffer)` overload and to pass `ValueStringBuilder.RawChars` for the second parameter within the loop?

Reply:
There are number of ways to write this. I went for consistency with what we do in the rest of `System.Environment`.